src/components/document/SubTabs/components/Files/Files.tsx

```diff
@@ -33,6 +33,13 @@ export const Files = observer(() => {
         setFetchingChildrenPage(false)
     }
 
+    useEffect(() => {
```

Comment:
why is it required?

Reply:
Without this `useEffect` hook, the internal state of `files`, `currentPage`, and `currentHasNextPage` would only be set once on component initialization and would not reflect any changes in `data` that occur afterwards. This can lead to stale or incorrect data being displayed in the component.

So, the main purpose of this `useEffect` is to keep the component's internal `state` in sync with changes to its `data` observable prop.

Reply:
but that's the exact thing that MobX is intended to do
by wrapping component in `observer` and making stores observable updates will be done automatically

Reply:
I understand your point that MobX is meant to handle state updates, and it indeed does when it comes to reactive rendering due to changes in the observable state.

However, in this case, we are using local state variables with `useState` which are initialized based on our data observable. These local state variables (`files`, `currentPage`, `currentHasNextPage`) do not automatically update even when data changes. observer allows our component to rerender when `data` changes, but it doesn't automatically sync the changes from data to our local state variables.

The `useEffect` hook is thus necessary to keep these local state variables in sync with our data observable. When `data` changes, the `useEffect` triggers and updates our local state accordingly. If we were to remove useEffect, we'd lose this sync and our local state would not reflect changes in data.

I hope this clarifies the purpose of `useEffect` in this instance."

Reply:
yeah, ok, got it, I thought that state was variables were removed
perfectly it would be to create a separate store for this component and keep all state management there but for now let's leave it like it is

Reply:
Yep, I agree, we can create some tasks to slowly move all state to the store and so on. But for the sake of this fix, I did not want to enter in to much refactoring. But yes, we will create new store in some other task :)